src/modules/job-manager/restart.h

```diff
@@ -20,6 +20,8 @@ int restart_from_kvs (struct job_manager *ctx);
 /* exposed for unit testing only */
 int restart_count_char (const char *s, char c);
 
+int checkpoint_to_txn (struct job_manager *ctx, flux_kvs_txn_t *txn);
```

Comment:
maybe it's just me, but this sounds you're checkpointing into the txn?  which makes no sense.  Hmmm.

`save/write/add_checkpoint_to_txn()`?  I'd say add is the best of this bunch.
`txn_checkpoint_max_jobid()`?

maybe there's a better one.

Reply:
Yeah I wasn't too happy with the function names either.  OK, I tacked on an earlier commit that renames `checkpoint_to_kvs()` to `restart_save_state()` and then made the new function `restart_save_state_to_txn()`.  Hopefully that works a little better, and also normalizes all the public functions in restart.c to have a `restar_*` prefix, and avoids the use of "checkpoint" which recently has become a bit of a loaded term.